futaba/cogs/settings/core.py

```diff
@@ -37,4 +43,234 @@ class Settings:
     def __init__(self, bot):
         self.bot = bot
 
-    # TODO
+        for guild in bot.guilds:
+            bot.sql.settings.get_special_roles(guild)
+
+    @commands.command(name='prefix')
+    async def prefix(self, ctx, *, prefix: str = None):
+        '''
+        Gets the current prefix. If you're a moderator, you can set it too.
+        A trailing underscore is converted into spaces. A single '_' unsets
+        the bot's prefix, and uses the default one.
+        '''
+
+        # Get prefix
+        if prefix is None:
+            bot_prefix = self.bot.prefix(ctx.message)
+            embed = discord.Embed(colour=discord.Colour.dark_teal())
+            if ctx.guild is None:
+                embed.description = 'No command prefix, all messages are commands.'
+            else:
+                embed.description = f'Prefix for {ctx.guild.name} is `{bot_prefix}`.'
+            reaction = Reactions.SUCCESS
+
+        # Attempt to set prefix outside of guild
+        elif ctx.guild is None and prefix is not None:
```

Comment:
Instead of this there is a decorator you can use `@commands.guild_only()` which will only allow the command to be run in a guild. Then we can use a custom error to display the error message to the user

Reply:
Well, the bot can be used in PMs, so it's not really "guild-only". I think being able to see that the prefix is nothing makes sense. I suppose we could use an exception to handle that case, but I don't believe it's an error.

Reply:
Okay fair